src/browserServerImpl.ts

```diff
@@ -78,16 +84,26 @@ export class BrowserServerLauncherImpl implements BrowserServerLauncher {
     return browserServer;
   }
 
-  private _onConnect(browser: Browser, scope: DispatcherScope, forceDisconnect: () => void) {
+  private _onConnect(browser: Browser, portForwardingServer: BrowserServerPortForwardingServer, scope: DispatcherScope, forceDisconnect: () => void) {
     const selectors = new Selectors();
     const selectorsDispatcher = new SelectorsDispatcher(scope, selectors);
     const browserDispatcher = new ConnectedBrowserDispatcher(scope, browser, selectors);
     browser.on(Browser.Events.Disconnected, () => {
       // Underlying browser did close for some reason - force disconnect the client.
       forceDisconnect();
     });
-    new PlaywrightDispatcher(scope, this._playwright, selectorsDispatcher, browserDispatcher);
+    const playwrightDispatcher = new PlaywrightDispatcher(scope, this._playwright, selectorsDispatcher, browserDispatcher, (ports: number[]) => {
+      portForwardingServer.enablePortForwarding(ports);
+    });
+    const incomingSocksSocketHandler = (socket: SocksSocket) => {
+      playwrightDispatcher._dispatchEvent('incomingSocksSocket', { socket: new SocksSocketDispatcher(playwrightDispatcher._scope, socket) });
+    };
+    if (portForwardingServer.enabled)
```

Comment:
I don't think we need to check for enabled here, portForwardingServer does it. Same below.

Reply:
without that `should be able to connect 20 times to a single server without warnings` resulted in a MaxListenersExceededWarning error. Maybe we should just disable the maxListeners in it.

Reply:
Just do `setMaxListeners(0)` on the port forwarding server.